### PR TITLE
fix: adapter configuration fixes

### PR DESCRIPTION
**Pull request by lekhovitsky**
Fixes:
* https://github.com/spearbit-audits/review-gearbox/issues/61
* https://github.com/spearbit-audits/review-gearbox/issues/63
* https://github.com/spearbit-audits/review-gearbox/issues/102
* https://github.com/spearbit-audits/review-gearbox/issues/110

Acknowledges:
* https://github.com/spearbit-audits/review-gearbox/issues/55
* https://github.com/spearbit-audits/review-gearbox/issues/103

### CODE REVIEW

File: contracts/credit/CreditConfiguratorV3.sol [hunk 1]
```diff
@@ -277,10 +277,10 @@ contract CreditConfiguratorV3 is ICreditConfiguratorV3, ACLNonReentrantTrait {
             revert AddressIsNotContractException(targetContract); // I:[CC-10A]
         }
 
-        if (
-            targetContract == creditManager || targetContract == creditFacade() || adapter == creditManager
-                || adapter == creditFacade()
-        ) revert TargetContractNotAllowedException(); // I:[CC-10C]
+        address cf = creditFacade();
+        if (targetContract == creditFacade() || adapter == creditFacade()) {
```
Comment (lekhovitsky) [inline contracts/credit/CreditConfiguratorV3.sol:281]: ah, `cf` is introduced but not used lol, fixed in https://github.com/Gearbox-protocol/core-v3/pull/279/commits/564f07daf0e8c3bb20ae118646fc2909999ddac9 (part of https://github.com/Gearbox-protocol/core-v3/pull/279)

**Comment by StErMi** [comment]
@lekhovitsky 

`allowAdapter` natspec should be updated because now the function does not (or at least not directly) reverts if `adapter` or `adapter`'s `targetContract` are equal to the `creditManager` because this check is performed directly inside the `CreditManagerV3(creditManager).contractToAdapter(...)` call

**Comment by cryptarasecurity** [comment]
> @lekhovitsky
> 
> `allowAdapter` natspec should be updated because now the function does not (or at least not directly) reverts if `adapter` or `adapter`'s `targetContract` are equal to the `creditManager` because this check is performed directly inside the `CreditManagerV3(creditManager).contractToAdapter(...)` call

I do agree that natspec could be updated to at least state that the check is performed indireclty as the check is now only performed inside the `CreditManagerV3(creditManager).setContractAllowance` function which does follow the recommendation of issue https://github.com/spearbit-audits/review-gearbox/issues/61 by removing the check under the configurator due to duplications on the `CreditManagerV3`.

**Comment by lekhovitsky** [comment]
@StErMi @cryptarasecurity It's fine for user-facing contracts like credit configurator or credit facade to make comments about checks or other actions performed in the credit manager, so decided not to update the comment.

**Comment by StErMi** [comment]
@lekhovitsky the `TargetContractNotAllowedException` used in `allowAdapter` is not entirely correct because the function reverts if the target contract OR **the adapter** are equal to the `CreditFacade`. The same would be for the `setCreditFacade` function.

**Comment by StErMi** [comment]
For the rest, the PR seems to fix/address all the linked issues linked above.

**Comment by lekhovitsky** [comment]
@StErMi Exception names are quite a mess but fixing them all consistently again creates a huge diff which nobody would want to review, so keeping those as is

**Pull request merged**